xml/http_servers.xml

```diff
@@ -1509,10 +1585,69 @@
 
   <fingerprint pattern="^SAP J2EE Engine/(\d+\.\d+)$">
     <description>SAP NetWeaver Web AS (Application Server)</description>
+    <example service.version="7.01">SAP J2EE Engine/7.01</example>
+    <param pos="0" name="service.vendor" value="SAP"/>
+    <param pos="0" name="service.product" value="NetWeaver Application Server"/>
+    <param pos="0" name="service.family" value="NetWeaver"/>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:sap:netweaver_application_server:{service.version}"/>
+  </fingerprint>
+
+  <fingerprint pattern="^SAP NetWeaver Application Server$">
+    <description>SAP NetWeaver Application Server without version</description>
+    <example>SAP NetWeaver Application Server</example>
+    <param pos="0" name="service.vendor" value="SAP"/>
+    <param pos="0" name="service.product" value="NetWeaver Application Server"/>
+    <param pos="0" name="service.family" value="NetWeaver"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:sap:netweaver_application_server:-"/>
+  </fingerprint>
+
+  <fingerprint pattern="^SAP NetWeaver Application Server ([\d.]+) / AS Java ([\d.]+)$">
+    <description>SAP NetWeaver Application Server Java</description>
+    <example service.version="7.22" service.component.version="7.30">SAP NetWeaver Application Server 7.22 / AS Java 7.30</example>
+    <param pos="0" name="service.vendor" value="SAP"/>
+    <param pos="0" name="service.product" value="NetWeaver Application Server Java"/>
+    <param pos="0" name="service.family" value="NetWeaver"/>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:sap:netweaver_application_server_java:{service.version}"/>
+    <param pos="2" name="service.component.version"/>
+  </fingerprint>
+
+  <fingerprint pattern="^SAP NetWeaver Application Server ([\d.]+) / ICM ([\d.]+)$">
+    <description>SAP NetWeaver Application Server - Internet Communication Manager</description>
+    <example service.version="7.21" service.component.version="7.21">SAP NetWeaver Application Server 7.21 / ICM 7.21</example>
     <param pos="0" name="service.vendor" value="SAP"/>
-    <param pos="0" name="service.product" value="NetWeaver Web AS"/>
+    <param pos="0" name="service.product" value="NetWeaver Internet Communication Manager"/>
     <param pos="0" name="service.family" value="NetWeaver"/>
     <param pos="1" name="service.version"/>
+    <param pos="2" name="service.component.version"/>
```

Comment:
Should there be `service.component.product` elements too?  Same for the above fingerprint too

Reply:
Addressing this in PR: https://github.com/rapid7/recog/pull/274